[2024-12-26T16:57:54.038+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-12-26T16:57:55.725+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: movie_pipeline.create_run_dir manual__2024-12-26T15:02:05.416978+00:00 [queued]>
[2024-12-26T16:57:55.758+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: movie_pipeline.create_run_dir manual__2024-12-26T15:02:05.416978+00:00 [queued]>
[2024-12-26T16:57:55.760+0000] {taskinstance.py:2866} INFO - Starting attempt 5 of 5
[2024-12-26T16:57:55.809+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): create_run_dir> on 2024-12-26 15:02:05.416978+00:00
[2024-12-26T16:57:55.824+0000] {standard_task_runner.py:72} INFO - Started process 3544 to run task
[2024-12-26T16:57:55.829+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'movie_pipeline', 'create_run_dir', 'manual__2024-12-26T15:02:05.416978+00:00', '--job-id', '41', '--raw', '--subdir', 'DAGS_FOLDER/pipeline_dag.py', '--cfg-path', '/tmp/tmpq_1uxx79']
[2024-12-26T16:57:55.842+0000] {standard_task_runner.py:105} INFO - Job 41: Subtask create_run_dir
[2024-12-26T16:57:56.462+0000] {task_command.py:467} INFO - Running <TaskInstance: movie_pipeline.create_run_dir manual__2024-12-26T15:02:05.416978+00:00 [running]> on host 1712831d7ea4
[2024-12-26T16:57:56.776+0000] {abstractoperator.py:783} ERROR - Exception rendering Jinja template for task 'create_run_dir', field 'op_args'. Template: ['{{logical_date-timedelta(days=2)}}']
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/abstractoperator.py", line 775, in _do_render_template_fields
    rendered_content = self.render_template(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/template/templater.py", line 183, in render_template
    return [self.render_template(element, context, jinja_env, oids) for element in value]
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/template/templater.py", line 183, in <listcomp>
    return [self.render_template(element, context, jinja_env, oids) for element in value]
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/template/templater.py", line 171, in render_template
    return self._render(template, context)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/abstractoperator.py", line 730, in _render
    return super()._render(template, context, dag=dag)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/template/templater.py", line 127, in _render
    return render_template_to_string(template, context)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/helpers.py", line 301, in render_template_to_string
    return render_template(template, cast(MutableMapping[str, Any], context), native=False)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/helpers.py", line 296, in render_template
    return "".join(nodes)
  File "<template>", line 13, in root
  File "/home/airflow/.local/lib/python3.9/site-packages/jinja2/sandbox.py", line 392, in call
    if not __self.is_safe_callable(__obj):
  File "/home/airflow/.local/lib/python3.9/site-packages/jinja2/sandbox.py", line 276, in is_safe_callable
    getattr(obj, "unsafe_callable", False) or getattr(obj, "alters_data", False)
  File "/home/airflow/.local/lib/python3.9/site-packages/jinja2/runtime.py", line 864, in __getattr__
    return self._fail_with_undefined_error()
  File "/home/airflow/.local/lib/python3.9/site-packages/jinja2/runtime.py", line 857, in _fail_with_undefined_error
    raise self._undefined_exception(self._undefined_message)
jinja2.exceptions.UndefinedError: 'timedelta' is undefined
[2024-12-26T16:57:56.815+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 3115, in _execute_task_with_callbacks
    task_orig = self.render_templates(context=context, jinja_env=jinja_env)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 3534, in render_templates
    original_task.render_template_fields(context, jinja_env)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/baseoperator.py", line 1438, in render_template_fields
    self._do_render_template_fields(self, self.template_fields, context, jinja_env, set())
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/abstractoperator.py", line 775, in _do_render_template_fields
    rendered_content = self.render_template(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/template/templater.py", line 183, in render_template
    return [self.render_template(element, context, jinja_env, oids) for element in value]
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/template/templater.py", line 183, in <listcomp>
    return [self.render_template(element, context, jinja_env, oids) for element in value]
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/template/templater.py", line 171, in render_template
    return self._render(template, context)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/abstractoperator.py", line 730, in _render
    return super()._render(template, context, dag=dag)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/template/templater.py", line 127, in _render
    return render_template_to_string(template, context)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/helpers.py", line 301, in render_template_to_string
    return render_template(template, cast(MutableMapping[str, Any], context), native=False)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/helpers.py", line 296, in render_template
    return "".join(nodes)
  File "<template>", line 13, in root
  File "/home/airflow/.local/lib/python3.9/site-packages/jinja2/sandbox.py", line 392, in call
    if not __self.is_safe_callable(__obj):
  File "/home/airflow/.local/lib/python3.9/site-packages/jinja2/sandbox.py", line 276, in is_safe_callable
    getattr(obj, "unsafe_callable", False) or getattr(obj, "alters_data", False)
  File "/home/airflow/.local/lib/python3.9/site-packages/jinja2/runtime.py", line 864, in __getattr__
    return self._fail_with_undefined_error()
  File "/home/airflow/.local/lib/python3.9/site-packages/jinja2/runtime.py", line 857, in _fail_with_undefined_error
    raise self._undefined_exception(self._undefined_message)
jinja2.exceptions.UndefinedError: 'timedelta' is undefined
[2024-12-26T16:57:56.886+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=movie_pipeline, task_id=create_run_dir, run_id=manual__2024-12-26T15:02:05.416978+00:00, execution_date=20241226T150205, start_date=20241226T165755, end_date=20241226T165756
[2024-12-26T16:57:56.936+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-12-26T16:57:56.938+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 41 for task create_run_dir ('timedelta' is undefined; 3544)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 3115, in _execute_task_with_callbacks
    task_orig = self.render_templates(context=context, jinja_env=jinja_env)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 3534, in render_templates
    original_task.render_template_fields(context, jinja_env)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/baseoperator.py", line 1438, in render_template_fields
    self._do_render_template_fields(self, self.template_fields, context, jinja_env, set())
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/abstractoperator.py", line 775, in _do_render_template_fields
    rendered_content = self.render_template(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/template/templater.py", line 183, in render_template
    return [self.render_template(element, context, jinja_env, oids) for element in value]
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/template/templater.py", line 183, in <listcomp>
    return [self.render_template(element, context, jinja_env, oids) for element in value]
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/template/templater.py", line 171, in render_template
    return self._render(template, context)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/abstractoperator.py", line 730, in _render
    return super()._render(template, context, dag=dag)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/template/templater.py", line 127, in _render
    return render_template_to_string(template, context)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/helpers.py", line 301, in render_template_to_string
    return render_template(template, cast(MutableMapping[str, Any], context), native=False)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/helpers.py", line 296, in render_template
    return "".join(nodes)
  File "<template>", line 13, in root
  File "/home/airflow/.local/lib/python3.9/site-packages/jinja2/sandbox.py", line 392, in call
    if not __self.is_safe_callable(__obj):
  File "/home/airflow/.local/lib/python3.9/site-packages/jinja2/sandbox.py", line 276, in is_safe_callable
    getattr(obj, "unsafe_callable", False) or getattr(obj, "alters_data", False)
  File "/home/airflow/.local/lib/python3.9/site-packages/jinja2/runtime.py", line 864, in __getattr__
    return self._fail_with_undefined_error()
  File "/home/airflow/.local/lib/python3.9/site-packages/jinja2/runtime.py", line 857, in _fail_with_undefined_error
    raise self._undefined_exception(self._undefined_message)
jinja2.exceptions.UndefinedError: 'timedelta' is undefined
[2024-12-26T16:57:57.005+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-12-26T16:57:57.100+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-12-26T16:57:57.112+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
